src/NetTopologySuite.IO.ShapeFile/Streams/ManagedStreamProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace NetTopologySuite.IO.Streams
+{
+    /// <summary>
+    /// A stream provider that provides <see cref="Stream"/>s who are managed by the application.
+    /// </summary>
+    public class ManagedStreamProvider : IStreamProvider
+    {
+        /// <summary>
+        /// Creates an instance of this class
+        /// </summary>
+        /// <param name="kind">The kind of stream</param>
+        /// <param name="stream">A <see cref="Stream"/> managed by the application</param>
+        public ManagedStreamProvider(string kind, Stream stream)
+        {
+            Kind = kind;
+            Stream = stream;
+        }
+
+        /// <summary>
+        /// Gets a value indicating that the underlying stream is read-only
+        /// </summary>
+        public bool UnderlyingStreamIsReadonly => !Stream.CanWrite;
+
+        protected Stream Stream { get; private set; }
+
+        /// <summary>
+        /// Function to return a Stream of the bytes
+        /// </summary>
+        /// <returns>An opened stream</returns>
+        public Stream OpenRead()
+        {
+            return Stream;
+        }
+
+        /// <summary>
+        /// Function to open the underlying stream for writing purposes
+        /// </summary>
+        /// <remarks>If <see cref="UnderlyingStreamIsReadonly"/> is not <value>true</value>
+        /// this method shall fail</remarks>
+        /// <returns>An opened stream</returns>
+        /// <exception cref="InvalidOperationException">Thrown if <see cref="UnderlyingStreamIsReadonly"/> is <value>true</value></exception>
+        public Stream OpenWrite(bool truncate)
```

Comment:
can you please add some documentation to this class that indicates what responsibilities this class imposes upon the caller to ensure correct operation, given that we ignore the value of the `truncate` parameter?

Reply:
Added implementation for truncate